metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -409,6 +409,21 @@ class Compilers(
     }
   }.getOrElse(Future.successful(Nil.asJava))
 
+  def extractMethod(
+      params: TextDocumentPositionParams,
+      applRange: Integer,
+      lv: Integer,
```

Comment:
We could send in the position to insert it into maybe? Wouldn't require making sure the lvl is the same in the compiler.